packages/cli/src/commands/build.ts

```diff
@@ -312,6 +313,43 @@ export default async function main(client: Client): Promise<number> {
   // TODO: parallelize builds
   const buildResults: Map<Builder, BuildResult> = new Map();
   const overrides: PathOverride[] = [];
+  const repoRootPath = cwd === workPath ? undefined : cwd;
+  const rootPackageJsonPath = repoRootPath || workPath;
+
+  if (process.env.ENABLE_EXPERIMENTAL_COREPACK === '1') {
+    const pkg = await readJSONFile<PackageJson>(
+      join(rootPackageJsonPath, 'package.json')
+    );
+    if (pkg instanceof CantParseJSONFile) {
+      console.warn(
+        'Warning: Could not enable corepack because package.json is invalid JSON'
+      );
+    } else if (!pkg?.packageManager) {
+      console.warn(
+        'Warning: Could not enable corepack because package.json is missing "packageManager" property'
+      );
+    } else {
+      console.log(
+        `Detected ENABLE_EXPERIMENTAL_COREPACK=1 and "${pkg.packageManager}" in package.json`
+      );
+      const cacheDir = join(
+        rootPackageJsonPath,
+        '.vercel',
+        'cache',
+        'corepack'
+      );
+      await fs.mkdirp(cacheDir);
+      process.env.COREPACK_HOME = join(cacheDir);
+      process.env.DEBUG = 'corepack';
```

Comment:
Should we set this on the `spawnAsync()` opts instead of globally?

Reply:
Globally is much safer which is why I put it in `vc build` instead of build-utils.

Basically, we want everything after this point to use corepack.

However we might need to unset after `vc build` runs.

During my testing it didn't seem like I needed to unset it (perhaps there is a child process for build??)

Reply:
Here's an example where pnpm7 is detected in package.json but after `vc build` runs, it reverts to the original/global pnpm6

<img width="565" alt="image" src="https://user-images.githubusercontent.com/229881/170520321-eac6ea7e-796e-489a-830f-38fb1b7e2990.png">

Reply:
Now that we have #7876 ,  I'll add a similar integration test so we can verify this is working as expected 🔜

Reply:
I added a `corepack disable` after the build completes, but I'm worried this is actually worse due to https://github.com/nodejs/corepack/issues/112

Reply:
My point was more about if someone is setting `DEBUG` to something else because they use it in their own build script, then we'd be overwriting it here. Maybe let's just augment the variable if it's already set - you can add multiple namespaces to the `DEBUG` env var separated by colon `:`.

Reply:
@TooTallNate Do you know if the `:` delimiter should be `;` on windows? I can't find any docs on that for debug 🤔

Reply:
Whoops, sorry, looks like it's comma-delimited `,`. Same on Windows.

Reply:
@TooTallNate That explains why I couldn't find `:` or `:` anywhere 🤣 

Ok updated in [b15fb49](https://github.com/vercel/vercel/pull/7871/commits/b15fb49b9ed94cebe7223265291e6213abd4cedf)